extensions/amp-accordion/1.0/amp-accordion.js

```diff
@@ -126,20 +126,31 @@ const bindSectionShimToElement = (element) => SectionShim.bind(null, element);
  * @param {!AccordionDef.HeaderProps} props
  * @return {PreactDef.Renderable}
  */
-function HeaderShim(sectionElement, {onClick}) {
+function HeaderShim(sectionElement, props) {
+  const {onClick} = props;
+  const ariaControls = props['aria-controls'];
   const headerElement = sectionElement.firstElementChild;
   useLayoutEffect(() => {
     if (!headerElement || !onClick) {
       return;
     }
     headerElement.addEventListener('click', onClick);
+    const hasTabIndex = headerElement.hasAttribute('tabindex');
+    if (!hasTabIndex) {
+      headerElement.setAttribute('tabindex', 0);
+    }
+    headerElement.setAttribute('aria-controls', ariaControls);
+    headerElement.setAttribute('role', 'button');
     if (sectionElement[SECTION_POST_RENDER]) {
       sectionElement[SECTION_POST_RENDER]();
     }
     return () => {
       headerElement.removeEventListener('click', devAssert(onClick));
+      if (!hasTabIndex) {
```

Comment:
No need to reset attributes. We only clear out event handlers because they could cause memory leaks.

Reply:
Got it, removed resetting of tabIndex.  For `tabIndex`, I also check if it already exists in case the user has specified a different `tabIndex` than 0.

Do you think we should do something like this for the `role` attribute.  Also the `aria-controls` attribute?  The `aria-controls` attribute (here in the header) matches the `id` attribute in the content, which means the `id` in content is automatically overwritten.  I was not sure how to get around this.  I think there is some risk in case there is CSS (or other things) attached to that id.  What do you think?

Reply:
I think `tabindex` protection is definitely a good idea. But I'd not worry about any other. However, the 0.1 is probably a good source of data for this. Is it updated there?

Reply:
In 0.1 `tabindex` is protected similar to this implementation.  `role` is not protected similar to this implementation.

`aria-controls` is done like this.  First the code checks for an `id` in the content node.  If `id` attribute exists, it will take that `id` and assign it to `aria-controls` of the corresponding header node.  If no `id` attribute exists, it will generate one (similar to how it's done in this implementation) and assign it to the `id` of the content section and the `aria-controls` of the corresponding header.